src/compiler/checker.ts

```diff
@@ -7200,11 +7312,17 @@ module ts {
             return result;
         }
 
-        function checkExpressionCached(node: Expression, contextualMapper?: TypeMapper): Type {
+        function checkExpressionCached(node: Expression, contextualMapper?: TypeMapper, forAwait?: boolean): Type {

```

Comment:
Please document `forAwait` - I can't tell if this is for an `await` in a for-loop, or if it's meant to be read as "check Expression...for await expressions`.